src/crimson/osd/pg_recovery.cc

```diff
@@ -40,10 +42,13 @@ PGRecovery::interruptible_future<bool>
 PGRecovery::start_recovery_ops(
   RecoveryBackend::RecoveryBlockingEvent::TriggerI& trigger,
   PglogBasedRecovery &recover_op,
-  size_t max_to_start)
+  size_t max)
 {
   assert(pg->is_primary());
   assert(pg->is_peered());
+  auto max_to_start = std::max(
+    max,
+    crimson::common::get_conf<uint64_t>("crimson_max_active_push_per_reactor"));
```

Comment:
I think that it's confusing that we support both `osd_recovery_max_single_start` and `crimson_max_active_push_per_reactor`.

My suggestion would be to divide `osd_recovery_max_single_start` by `smp::count` to get the number of active pushes allowed for each reactor instead of introducing the new opiton.

```
      return pg->get_recovery_handler()->start_recovery_ops(
	trigger,
	*this,
	crimson::common::local_conf()->osd_recovery_max_single_start);
    });
```
```
PGRecovery::start_recovery_ops(
  RecoveryBackend::RecoveryBlockingEvent::TriggerI& trigger,
  PglogBasedRecovery &recover_op,
  size_t osd_max_to_start)
{
  ..
  auto max_active_pushes_per_reactor = osd_max_to_start / seastar::smp::count
  DEBUG(...)
```

What do you think?

Reply:
yeah, will do.

Reply:
@Matan-B Hi, matan, after a second thought, `max_active_pushes_per_reactor` limits the active pushes like `osd_recovery_max_active` does, so maybe `auto max_active_pushes_per_reactor = osd_recovery_max_active / seastar::smp::count` is better? What do you think?

Reply:
> Seems correct.

I'm also trying to refactor this PR, changing `OperationThrottle` to work at the `push/pull` level as opposed to the current way, limiting the active recoveries/pushes for both pglog-based recoveries and backfills in the same shard. What do you think?

Reply:
It does make sense to introduce both `osd_recovery_max_single_start` and `osd_recovery_max_active` in the same go as their logic is somewhat intertwined (`OSDService::_maybe_queue_recovery()`)

Is it a follow up to Mohit's https://github.com/ceph/ceph/pull/62238 or a different version of it?

Reply:
Um, I mean that, at present, the `OperationThrottler` is used to throttle the pglog-based recovery operation which can contain multiple object recoveries, as shown in the code: https://github.com/ceph/ceph/blob/7861012264b7dc9062f08e37d6b32ab8c88fdae6/src/crimson/osd/osd_operations/background_recovery.cc#L80-L91

I intend to turn to use `OperationThrottler` to throttle the number of active recoveries instead of recovery operations. I think this is orthogonal to #62238 , am I right?

Reply:
I think so:-)